python/tank/settings/user.py

```diff
@@ -0,0 +1,246 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+User settings management.
+"""
+
+from __future__ import with_statement
+
+import os
+import ConfigParser
+import threading
+
+from .errors import MissingConfigurationFileError
+from .. import LogManager
+
+logger = LogManager.get_logger(__name__)
+
+class Singleton(object):
```

Comment:
I'm probably over-designing this. This is a thread-safe class that you can derive from to write a single class that is lazily initialized. I noticed the LogManager was also written this way, but not in a thread-safe manner. What do you think @manneohrstrom ?